src/lua/error.lua

```diff
@@ -278,13 +268,16 @@ local function error_to_string(err)
         return error_message(err)
     end
 
-    local cur = err
-    local error_stack = {}
-    while cur ~= nil do
-        table.insert(error_stack, 1, error_to_string_wo_prev(cur))
-        cur = cur.prev
+    local ibuf = buffer.internal.cord_ibuf_take()
+    local len = ffi.C.error_to_string(err, ibuf.wpos, ibuf:capacity())
```

Comment:
I think that we should mention in the changelog that we changed the way `tostring()` formats errors even though the new behavior is enabled in compat. I'd also do this in a separate commit.